src/main/java/org/example/Main.java

```diff
@@ -122,338 +163,205 @@ public boolean isHappy() {
-                        }
-                        if (users.contains("JUMPING") && !num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("HAPPY") && !num.happy) {
-                            continue;
-                        }
-                        if (users.contains("SAD") && !num.sad) {
-                            continue;
-                        }
-                        if (users.contains("-EVEN") && num.even) {
-                            continue;
-                        }
-                        if (users.contains("-ODD") && num.odd) {
-                            continue;
-                        }
-                        if (users.contains("-BUZZ") && num.buzz) {
-                            continue;
-                        }
-                        if (users.contains("-DUCK") && num.duck) {
-                            continue;
-                        }
-                        if (users.contains("-PALINDROMIC") && num.palindromic) {
-                            continue;
-                        }
-                        if (users.contains("-SPY") && num.spy) {
-                            continue;
-                        }
-                        if (users.contains("-GAPFUL") && num.gapful) {
-                            continue;
-                        }
-                        if (users.contains("-SUNNY") && num.sunny) {
-                            continue;
-                        }
-                        if (users.contains("-SQUARE") && num.square) {
-                            continue;
-                        }
-                        if (users.contains("-JUMPING") && num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("-HAPPY") && num.happy) {
-                            continue;
-                        }
-                        if (users.contains("-SAD") && num.sad) {
-                            continue;
-                        }
-                        num.printShortInfo();
-                        count--;
-                    }
+    private static boolean hasExclusiveProperty(Collection<String> users) {
+        if (users.isEmpty()) {
+            return false;
+        }
+        // TODO how to import pair
+        String[][] names = {
+                {"EVEN", "ODD"},
+                {"-EVEN", "-ODD"},
+                {"DUCK", "SPY"},
+                {"SQUARE", "SUNNY"},
+                {"HAPPY", "SAD"},
+                {"-HAPPY", "-SAD"},
+
+        };
+
+        if (users.size() > 0) {
+            boolean flag = false;
+            for (var name : names) {
+                if (users.contains(name[0]) && users.contains(name[1])) {
+                    System.out.println("The request contains mutually exclusive properties: [" + name[0] + ", " + name[1] + "].");
+                    flag = true;
```

Comment:
You should break here and I think no need to do 2nd loop if you got flag = true?